src/main/java/core/basesyntax/Ball.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax;
+
+public class Ball {
+    private String color;
+    private int number;
+
+    public Ball(String color, int number) {
+        this.color = color;
+        this.number = number;
+    }
+
+    @Override
+    public String toString() {
+        return "Ball { "
+                + " color = "
+                + color
+                + ", number = "
+                + number
+                + " }";
+    }
+
+    public enum Color {
+        RED, BLUE, GREEN, YELLOW, ORANGE, PURPLE, PINK;
+        private static Color[] colors = Color.values();
+
+        public static Color getByIndex(int index) {
```

Comment:
Actually, there is no sense to create Color[] and getByIndex(int index) method 
see comment in ColorSupplier

Reply:
I created one static array so that a new one is not created every time the method is embedded